client/client.go

```diff
@@ -283,6 +292,24 @@ func (c Client) PutRoleMeta(domain string, roleName string, auditRef string, rol
 	return err
 }
 
+func (c Client) GetRoleMetaResourceState(roleMetaResourceState, requestedState int) bool {
+	return getResourceState(roleMetaResourceState, c.RoleMetaResourceState, requestedState)
+}
+
+func (c Client) GetGroupMetaResourceState(groupMetaResourceState, requestedState int) bool {
+	return getResourceState(groupMetaResourceState, c.GroupMetaResourceState, requestedState)
+}
+
+func getResourceState(resourceState, clientState, requestedState int) bool {
+	if resourceState == -1 {
+		resourceState = clientState
+	}
+	if resourceState == -1 {
```

Comment:
is it should be `requestedState` ?

Reply:
No, what we're doing here is if the resource state is not specified (-1), then we default to the client state. And if there is no state set - still -1, we just return false. The requested state will never be -1 since that's our internal check state and not coming from the user specified data.